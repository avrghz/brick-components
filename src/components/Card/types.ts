export const SHADOW = ['never', 'always', 'hover'] as const

export type Shadow = typeof SHADOW[number]
