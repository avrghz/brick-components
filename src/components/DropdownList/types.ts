export interface Option {
    label: string
    value: string
    disabled?: boolean
}

export type Callback = (index: number) => void
