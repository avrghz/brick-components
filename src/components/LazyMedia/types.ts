export type MediaType = 'img' | 'picture'
