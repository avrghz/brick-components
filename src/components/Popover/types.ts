export const TRIGGER_ON = ['click', 'focus', 'hover', 'manual'] as const

export type TriggerOn = typeof TRIGGER_ON[number]
