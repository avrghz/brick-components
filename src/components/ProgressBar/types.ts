export const BAR_TYPE = ['line', 'circle', 'dashboard'] as const

export type BarType = typeof BAR_TYPE[number]

export interface ProgressProps {
    progress: number
    showPercentageText?: boolean
    textInside?: boolean
}

export interface ProgressTextProps extends Pick<ProgressProps, 'progress'> {
    show?: boolean
    isInnerText?: boolean
}

export interface ProgressCircleProps extends ProgressProps {
    strokeDash: [number, number]
    strokeDashOffset: number
    flip?: boolean
}
