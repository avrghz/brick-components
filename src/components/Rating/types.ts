export interface StarProps {
    hover?: boolean
    activeColor?: string
}

export interface Colors {
    [key: number]: string
}
