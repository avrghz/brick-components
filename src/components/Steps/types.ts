import { JSXBase } from '@stencil/core/internal'

export const STEP_STATE = ['process', 'wait', 'success', 'error', 'finish'] as const

export interface StepComponent {
    step: number
    title: string
    description?: string
    state?: typeof STEP_STATE[number]
}

export interface StepProps extends StepComponent {
    style?: JSXBase.HTMLAttributes<HTMLDivElement>['style']
    isLast?: boolean
    isCentered?: boolean
}
