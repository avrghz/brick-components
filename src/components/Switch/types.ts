export interface AVChange {
    isOn: boolean
}
