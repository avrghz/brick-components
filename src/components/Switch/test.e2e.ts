import { newE2EPage, E2EPage, E2EElement } from '@stencil/core/testing'
import { VARIANTS } from '../../shared/types'

describe('<bk-switch/>', () => {
    let page: E2EPage
    let component: E2EElement

    const getCheckbox = async () => await page.find('bk-switch input[type=checkbox]')
    const getSwitch = async () => await page.find('bk-switch .bk-switch')

    beforeEach(async () => {
        page = await newE2EPage()
        await page.setContent(`<bk-switch/>`)
        component = await page.find('bk-switch')
    })

    it('should render', async () => {
        expect(component).not.toBeNull()
        expect(component).toMatchSnapshot()
    })

    it('should render as disabled', async () => {
        component.setProperty('disabled', true)
        await page.waitForChanges()
        const checkbox = await getCheckbox()
        expect(await checkbox.getProperty('checked')).not.toBeTruthy()
    })

    it('should fire bkChange on every state change', async () => {
        const bkChange = await component.spyOnEvent('bkChange')
        component.click()
        await page.waitForChanges()

        expect(bkChange).toHaveReceivedEventDetail({
            isOn: true,
        })

        component.click()
        await page.waitForChanges()

        expect(bkChange).toHaveReceivedEventDetail({
            isOn: false,
        })
    })

    it('should render as checked', async () => {
        component.setProperty('isOn', true)
        await page.waitForChanges()
        const checkbox = await getCheckbox()
        expect(await checkbox.getProperty('checked')).toBeTruthy()
    })

    describe('Prop: variant', () => {
        VARIANTS.forEach((v) => {
            it(`should render as ${v}`, async () => {
                component.setProperty('variant', v)
                await page.waitForChanges()
                expect(await getSwitch()).toHaveClass(`bk-switch--${v}`)
            })
        })
    })
})
