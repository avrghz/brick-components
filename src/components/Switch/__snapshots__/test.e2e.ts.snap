// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bk-switch/> should render 1`] = `
<bk-switch aria-label="switch" class="hydrated sc-bk-switch-h sc-bk-switch-s" role="switch" tabindex="0">
  <div class="bk-switch bk-switch--default sc-bk-switch">
    <input aria-hidden="true" class="bk-switch__input sc-bk-switch" name="chk" tabindex="-1" type="checkbox">
    <span class="bk-switch__core sc-bk-switch"></span>
  </div>
</bk-switch>
`;
